kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/OperatorChangeDetectionST.java

```diff
@@ -313,6 +361,14 @@ void setupBefore() {
     private String getInitialChecksum(String namespace) {
         var kubeClient = kubeClient(namespace);
         AtomicReference<String> checksumFromAnnotation = new AtomicReference<>();
+        await().atMost(Duration.ofSeconds(90))
+                .pollDelay(Duration.ofSeconds(10))
```

Comment:
what is this 10 second pollDelay for?

Reply:
delay is the initial delay I think, rather than the interval between polls:

```
Specify the delay that will be used before Awaitility starts polling for
     * the result the first time. If you don't specify a poll delay explicitly
     * it'll be the same as the poll interval.
```

so I think this will sit there 10 seconds before the first check

Reply:
wee test:

```java
System.out.println(Instant.now());
        Awaitility.await().pollDelay(Duration.of(5, ChronoUnit.SECONDS))
                .until(() -> {
                    System.out.println(Instant.now());
                    return false;
                });
```
                
```
2025-06-18T02:44:27.705585040Z // before
2025-06-18T02:44:32.726462269Z // 5 second delay
2025-06-18T02:44:32.829071885Z // low interval from here
2025-06-18T02:44:32.929639615Z
2025-06-18T02:44:33.030189681Z
2025-06-18T02:44:33.130766118Z
2025-06-18T02:44:33.231374614Z
2025-06-18T02:44:33.331910664Z
2025-06-18T02:44:33.432480620Z
```

Reply:
if the assertion block is blocking for a long time it wouldn't need a big poll interval either, else it'll go like:

```
waitUntilCondition blocks for 9 seconds
pollInterval waits for 10 seconds
waitUntilCondition blocks for 9 seconds
pollInterval waits for 10 seconds
...
```

Reply:
for example with some work taking longer than the interval

``` java
System.out.println("begin: " + Instant.now());
        Awaitility.await().pollInterval(Duration.of(1, ChronoUnit.SECONDS))
                .until(() -> {
                    System.out.println("do blocking stuff: " + Instant.now());
                    Thread.sleep(5000);
                    System.out.println("finish blocking stuff: " + Instant.now());
                    return false;
                });
```

```
begin: 2025-06-18T03:49:56.263186977Z
do blocking stuff: 2025-06-18T03:49:57.282290881Z
finish blocking stuff: 2025-06-18T03:50:02.282656123Z
do blocking stuff: 2025-06-18T03:50:03.286124811Z
...
```